services/blockstorage/sync/state_idle.go

```diff
@@ -0,0 +1,49 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/synchronization"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+)
+
+type idleState struct {
+	config      blockSyncConfig
+	logger      log.BasicLogger
+	restartIdle chan struct{}
+	sf          *stateFactory
+}
+
+func (s *idleState) name() string {
+	return "idle-state"
+}
+
+func (s *idleState) processState(ctx context.Context) syncState {
+	noCommitTimer := synchronization.NewTimer(s.config.BlockSyncNoCommitInterval())
+	select {
+	case <-noCommitTimer.C:
+		s.logger.Info("starting sync after no-commit timer expired")
+		return s.sf.CreateCollectingAvailabilityResponseState()
+	case <-s.restartIdle:
+		return s.sf.CreateIdleState()
+	case <-ctx.Done():
+		return nil
+	}
+}
+
+func (s *idleState) blockCommitted() {
+	select {
+	case s.restartIdle <- struct{}{}:
+		s.logger.Info("sync got new block commit")
+	default:
+		s.logger.Info("channel was not ready, skipping notification")
```

Comment:
please document why this default clause is required here

Reply:
good idea, bled over that one